app/src/main/java/com/example/campusguide/MapsActivity.kt

```diff
@@ -42,6 +44,22 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
             }
         }
 
+        val navigateButton = findViewById<FloatingActionButton>(R.id.navigateButton)
+        navigateButton.setOnClickListener {
+            val getDirectionsDialogFragment =
+                GetDirectionsDialogFragment(
+                    GetDirectionsDialogFragment.DirectionsDialogOptions(
+                        "Concordia University Hall Building",
+                        "Faubourg Building"
+                    ) { start, end ->
+                        //Display the directions time
```

Comment:
this is roughly where we'd put the remove() method from Route before displaying new directions right?

Reply:
Yeah, really we should store the route in a member variable so it can be cleared here before a new one is created.

Reply:
Route is now stored in a member variable, but also now it has a set(start, end) method for drawing a new route. The old route is cleared when set is called.